File: 2593. Find Score of an Array After Marking All Elements/Solution.py
import heapq

class Solution:
    def findScore(self, nums: List[int]) -> int:
        n, res = len(nums), 0
        pq = []
        for i in range(n): heapq.heappush(pq, (nums[i], i))
        marked = [False for i in range(n)]
        while pq:
            x, i = heapq.heappop(pq)
            if marked[i]: continue
            marked[i] = True
            res += x
            if i > 0: marked[i - 1] = True
            if i < n - 1: marked[i + 1] = True
        return res

File: 2593. Find Score of an Array After Marking All Elements/Solution.java
class Solution {
    public long findScore(int[] nums) {
        PriorityQueue<int[]> pq = new PriorityQueue<>((a, b) -> {
            if (a[0] == b[0]) return a[1] - b[1];
            return a[0] - b[0];
        });
        int n = nums.length;
        for (int i = 0; i < n; i++) pq.offer(new int[] { nums[i], i });
        boolean[] marked = new boolean[n];
        long res = 0;
        while (!pq.isEmpty()) {
            int[] curr = pq.poll();
            if (marked[curr[1]]) continue;
            res += curr[0];
            if (curr[1] > 0) marked[curr[1] - 1] = true;
            if (curr[1] < n - 1) marked[curr[1] + 1] = true;
            marked[curr[1]] = true;
        }
        return res;
    }
}